[Event Hubs Client] Core Samples Overhaul (#16479)
Azure/azure-sdk-for-net: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample03_EventHubMetadata.md

```C# Snippet:EventHubs_Sample03_InspectHub
var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
var eventHubName = "<< NAME OF THE EVENT HUB >>";

var producer = new EventHubProducerClient(connectionString, eventHubName);

try
{
    EventHubProperties properties = await producer.GetEventHubPropertiesAsync();

    Debug.WriteLine("The Event Hub has the following properties:");
    Debug.WriteLine($"\tThe path to the Event Hub from the namespace is: { properties.Name }");
    Debug.WriteLine($"\tThe Event Hub was created at: { properties.CreatedOn }, in UTC.");
    Debug.WriteLine($"\tThe following partitions are available: [{ string.Join(", ", properties.PartitionIds) }]");
}
finally
{
    await producer.CloseAsync();
}
```

# Query the partitions of an Event Hub

> Due to their importance, there is also a shorthand way to query the partitions of an Event Hub.  The ability to query the Event Hub properties is available using the `EventHubProducerClient` and `EventHubConsumerClient`.  For illustration, the `EventHubProducerClient` is demonstrated, but the concept and form are common across both clients.

Reviewer: ```suggestion
Due to their importance, there is also a shorthand way to query the partitions of an Event Hub.  This capability is available using the `EventHubProducerClient` and `EventHubConsumerClient`.  For illustration, the `EventHubProducerClient` is demonstrated, but the concept and form are common across both clients.
```

Author: Applying manually to avoid throwing off my rebase.

```C# Snippet:EventHubs_Sample03_QueryPartitions
var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
var eventHubName = "<< NAME OF THE EVENT HUB >>";

var producer = new EventHubProducerClient(connectionString, eventHubName);

try
{
    string[] partitions = await producer.GetPartitionIdsAsync();
    Debug.WriteLine($"The following partitions are available: [{ string.Join(", ", partitions) }]");
}
finally
{
    await producer.CloseAsync();
}
```

# Query the properties of a partition

Some application scenarios call for understanding the "backlog" of events in an Event Hub, where an event being consumed is compared with the latest event available in the partition to understand how many events have accumulated and have yet to be processed.  This allows applications to make reasoned decisions about scaling consumers or throttling publishers to ensure that event processing does not fall too far behind. To support this, the `EventHubProducerClient` and `EventHubConsumerClient` provide the ability to query partitions of the Event Hub for information about its current state.

> This query is useful for occasionally inspecting partitions, but should not be used frequently to avoid saturating the connection and impacting performance.  When this information is needed often, an option can be set when reading events to query the last published event information for a partition in real-time.  More detail can be found in [Sample05_ReadingEvents](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample05_ReadingEvents.md#query-partition-information-while-reading).

Reviewer: ```suggestion
This query is useful for occasionally inspecting partitions, but should not be used frequently as it may negatively impact performance by saturating the connection.  When this information is needed often, an option can be set when reading events to query the last published event information for a partition in real-time.  More detail can be found in [Sample05_ReadingEvents](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample05_ReadingEvents.md#query-partition-information-while-reading).
```

Author: Applying manually to avoid throwing off my rebase.
